### PR TITLE
Add API proxy lambda function

### PR DESCRIPTION
**Pull request by ChristienGuy**
Add api proxy lambda. Update start command to run lambda serve. Update app to use BASE_URL. Add proxy to all http requests.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,13 +10,15 @@
 
 # production
 /build
+/lambda
```
Comment (ChristienGuy) [inline .gitignore:13]: This is the folder that build lambda functions will live when deployed.

File: .gitignore [hunk 2]
```diff
@@ -10,13 +10,15 @@
...
 
 # misc
 .DS_Store
 .env.local
 .env.development.local
 .env.test.local
 .env.production.local
+.envrc
```
Comment (ChristienGuy) [inline .gitignore:21]: I use https://direnv.net/ for doing folder specific environment variables. So this is the folder that stores it's setup (including auth tokens so it's ignored)

File: netlify.toml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+[build]
```
Comment (ChristienGuy) [inline netlify.toml:1]: netlify.toml file is used to tell Netlify how to build the app and where the output will live

File: package.json [hunk 4]
```diff
@@ -1,17 +1,30 @@
 {
-  "name": "new-app-with-chris",
+  "name": "genuis",
```
Comment (ChristienGuy) [inline package.json:2]: seems only right.

File: package.json [hunk 5]
```diff
@@ -1,17 +1,30 @@
...
+    "netlify-lambda": "^1.0.3",
     "react-dom": "^16.6.3",
     "react-scripts": "2.1.1",
+    "react": "^16.6.3",
+    "serverless-http": "^1.8.0",
     "styled-components": "^4.1.1"
   },
+  "devDependencies": {
+    "http-proxy-middleware": "^0.19.1",
+    "npm-run-all": "^4.1.5"
```
Comment (ChristienGuy) [inline package.json:19]: These are used for local proxying and for build steps.

File: src/lambda/proxy.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+app.use(bodyParser.json());
+
+app.get('*', async (req, res) => {
+  const { originalUrl } = req;
+  const endpoint = originalUrl.replace('/proxy', '');
+
+  try {
+    const response = await axios.get(`https://api.genius.com${endpoint}`, {
+      headers: {
+        Authorization: `Bearer ${process.env.AUTH_TOKEN}`
```
Comment (ChristienGuy) [inline src/lambda/proxy.js:16]: AUTH_TOKEN now comes from environment variables so you'll need to set it up to work locally and will need to add it to Netlify for deployment.

File: src/setupProxy.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+const proxy = require('http-proxy-middleware');
+
+module.exports = function(app) {
+  app.use(proxy('/.netlify/functions/', {
```
Comment (ChristienGuy) [inline src/setupProxy.js:4]: Netlify lambda functions are made available as `/.netlify/functions/<filename>`.

So in order to get to the 'proxy' file we need to hit `/.netlify/functions/proxy/`.

This local proxy basically says any local calls to `/.netlify/functions/` should get redirected to `localhost:9000/.netlify/functions/`

**Review by ChristienGuy — COMMENTED** [review]
I've left a handful of comment around things I think are not particularly intuitive. I'm happy to talk through this another time tho if you want to better understand what's happening.

**Review by Jack-Gill — APPROVED** [review]

**Pull request merged**